### PR TITLE
Collections redesign

### LINKED ISSUE #13660: Collections Redesign Implementation
Tracking work for an update to collections to make it much easier to browse through collections. This update also includes some UI tweaks to collection content lists to make them easier to use.

**Links**
- [Product doc](https://docs.google.com/document/d/1LGLyhnCeAVzQ4wuaqgTKmOdkR4Gxb2nTge750N-XonA/edit)
- Feature branch: `0.38-collections-redesign`

**Implementation Plan**

**ETA**: 11/20/2020

- [x] Mock out new UI and components 
- [X] Update / add endpoint to return all collections a user can see in a nested structure (https://github.com/metabase/metabase/commit/8303be7675f3dffd760554de55481c76c98997b1)
- [x] Wire up endpoint and new collections sidebar (https://github.com/metabase/metabase/pull/13684)
- [x] Update collection page to use new components / integrate with sidebar (https://github.com/metabase/metabase/pull/13684)
- [x] Add expand / collapse actions (https://github.com/metabase/metabase/pull/13684)
- [x] Update visual design for collection content (pins, items)
- [x] Update visual design for collection sidebar 
- [x] Update admin personal collection browsing.
- [x] Fix issue where `ItemPicker` relies on old list endpoint structure for the collections entity api https://github.com/metabase/metabase/pull/13765/commits/e909deeb0f9361b9d2c4de6b0e432b2afee00cda
- [x] Visual design tweaks based on feedback https://github.com/metabase/metabase/pull/13822
- [x] Need to be able to move pinned items
- [x] Open ancestral collections in the sidebar when deep linking https://github.com/metabase/metabase/pull/13765/commits/5ed274e99a1a2df2bcfdd66577170bd4330710c3
- [x] Update sidebar drag and drop hover styles

**Tests**
- [x] Update test fixture data to have a few deeply nested collections with long names. -https://github.com/metabase/metabase/pull/13846
- [x] Fix up basic drag and drop between collection content and the sidebar
- [x] Update and expand collection related cypress tests.  https://github.com/metabase/metabase/pull/13846

### PR DESCRIPTION
**Pull request by kdoh**
Updates collections to be easier to browse with a new sidebar.

image

We've known for a bit that browsing through collections isn't always an easy experience. If you can't remember what folder something is in there's a lot of clicking back and forth and not much context to hold on to since so much of the layout changes. For 0.38, we're updating the experience to have a persistent sidebar showing all the collections you have access to. You can expand any collection that has sub collections to see what's inside and more quickly bounce between them to make the process of finding what you're looking for a little easier. Our hope is that the increased ease of moving around will make folks more likely to explore a bit more than they otherwise would.

Related changes:
- We've moved "new dashboard" and "new pulse" creation inside of collections from the main navigation. We've felt for a while that that main nav item felt a little bit out of place since dashboard creation isn't necessarily an every day thing.

Closes #13660

### CODE REVIEW

**Comment by kdoh** [comment]
@paulrosenzweig I think this is ready for a first pass code review.

File: frontend/src/metabase/collections/containers/CollectionContent.jsx
Comment (paulrosenzweig) [inline frontend/src/metabase/collections/containers/CollectionContent.jsx:36]: I can't figure out how to use these in the UI. It looks like bulk actions weren't affected by this change, but wanted to call that out anyways.

**Review by paulrosenzweig — APPROVED** [review]
Looks good. I like the new experience a lot!

**Comment by mazameli** [comment]
Here's feedback from a pretty detail-oriented, nitpicky pass through the branch.

After creating a sub-collection inside a collection, I'd expect to have the parent collection expand out in the sidebar so that I can see my new sub-collection:

new-subfolder

On a similar note, this scenario disorients me, where I click into a subfolder, then click on an item, then click the folder icon to go back, but then I sort of forget where I am because the path to the folder I'm in is collapsed when I'd expect it to be expanded:

subfolder-collapsed

I think we're subverting expectations here that clicking on the expanded chevron doesn't collapse that level down:

collapse-expectation

It would be cool to get some hover effect when doing a drag-to-move. (Incidentally, it doesn't look like I can drag and move folders in the sidebar yet.)

drag-move-affordance


It looks like something broke in the collections browser component where I now can't drill down into a folder to see its subfolders, e.g. the "Google Analytics Statistics" folder has subfolders, and so there should be a right arrow in a circle to the right of this element:

image

This layout still feels floaty to me when there are only a handful of items in a folder:

image

I don't know that this is quite working for me yet. I think part of it is that the icon colors are all different, and the shadows around the cards fade out completely at the top which makes them lose definition. The question icon to me also looks like the correct icon couldn't be found, like a broken resource link:

image

Hover state here seems to be missing padding:

image

Wonder if this would look better if the icon went blue on hover:

image

Do we want to switch out `text-dark` to `#4C5773` yet? The black used here feels a bit stock or off-the-rack to me, particularly on the icons:

image

There were some mocks you'd done where questions were the reverse of dashboards, brand-on-light-blue, and pinned items were yellow, with all the items' text in blue. That felt a bit more colorful and bright to me than what's in the branch:

image

I'm kind of wondering if this whole thing would feel cleaned up if we killed the folder icon. Everything in the sidebar is a folder, so it feels a bit redundant. I think you could keep the icons next to the personal collection(s) and archive though, since they're separated out from the folders.

image

How are you feeling about this top-right space? I agree that I don't think we want to duplicate the robot face here, but it feels slightly under baked. Do we need it? Maybe part of the problem is that the basline of the text isn't bottom-aligned with the baseline of the large page heading.

image

**Comment by salsakran** [comment]
"other user's personal collections" still goes to the old style display for me.

File: frontend/src/metabase/collections/containers/CollectionContent.jsx
Comment (kdoh) [reply frontend/src/metabase/collections/containers/CollectionContent.jsx:36]: Yeah the cue to enter bulk actions is too subtle and we're going to rework that. Right now you hover on the icon of the item and that lets you click a checkbox and that puts the view into multi select mode. It's unchanged from the old version in this new branch.

File: frontend/src/metabase/collections/containers/Greeting.jsx
Comment (kdoh) [reply frontend/src/metabase/collections/containers/Greeting.jsx:13]: I think I might end up removing this until we end up merging the overworld and the root collection. Feels a bit frankenstein'd until we do that.

File: frontend/src/metabase/collections/components/CollectionLink.jsx
Comment (kdoh) [reply frontend/src/metabase/collections/components/CollectionLink.jsx:15]: The negative margin and then padding based on the depth helps make nested items in the sidebar have full width hovers. There might be a cleaner way to do it but this works ok. I'll leave a comment in the code here about it though.

**Comment by mazameli** [comment]
Definitely liking the color tweaks to item icons! I think the stroke you added below the heading of collections where there aren't any pinned items helps with the floaty feeling I had, too.

Here are some smaller things I found:

------

Collapsing a folder structure by clicking on the chevron isn't working for me. Maybe that's in a separate PR that hasn't been merged in here yet? Seems like that's the case with some of the other folder expansion items I mentioned, so I'll hold off on that.

This is very nitpicky, but I'm pretty sure these guys are slightly ovoid rather than perfectly circular:

image

I'm still seeing that padding problem in Safari with "Our analytics." Safari on left, Chrome on right:

image

Looks like there's a bit more padding at the bottom of the content list than needed (that blue focus outline appears when I click on the empty area, incidentally):

image

**Comment by mkopinsky** [comment]
Quick thought from a user, spurred by @salsakran's comment "Folders can be empty because they're not yet filled or because they're being used merely to contain subfolders. The latter case feels marginalized.". (I realize this branch is merged, but I figured rather put the feedback close to that comment than create a new issue.)

It would be great if the sidebar showed pinned items in addition to subfolders. I sometimes find myself navigating through collections looking for a specific dashboard, not knowing whether I put it in the parent collection or nested somewhere in a sub-collection.

I might have a collection structure like this:
```
Project A
 └─ (pinned) Operational dashboard
 └─ (pinned) Analytics dashboard
 └─ Operational
     └─ ... various operational questions, some of which might be sent out via pulse
 └─ Analytics
     └─ ... various analytical graphs, which I would honestly never navigate to except by clicking through from the dashboard
Mega-project B
 └─ Sub-project A
     └─ (pinned) Operational dashboard
     └─ (pinned) Analytics dashboard
     └─ Operational
         └─ ... various operational questions, some of which might be sent out via pulse
     └─ Analytics
         └─ ... various analytical graphs, which I would honestly never navigate to except by clicking through from the dashboard
 └─ Sub-project B
     └─ (same stuff again)
```

I often put the dashboards within the parent collection so as to avoid an extra click/page load when navigating. This change with a folder structure in the sidebar is great, as it will save me/my users from having to click through so many pages to get to the dashboards. So 👍 !

However, this introduces a new "problem", as I now need to click on Project A (to see if the dashboards are in the top-level project collection) as well as the sub-collection, since I may not remember where I saved them. If the sidebar can show the sub-collection as well as the pinned dashboards, that would make it super-easy to find what I'm looking for.

It would be great if the sidebar could look something like this. (I hope you can excuse my unicode "mockup".)
```
📁 Project A
  📄 Operational dashboard
  📄 Analytics dashboard
  📁 Operational
  📁 Analytics
📁 Mega-project B
    📁 Sub-project A
       📄 Operational dashboard
       📄 Analytics dashboard
       📁 Operational
       📁 Analytics
   📁  Sub-project B
      ... etc
```

**Comment by mazameli** [comment]
Thanks for this feedback, @mkopinsky. Glad to hear you're liking the new sidebar overall. 

So are you saying that you'd like those pins to show up in the tree to prevent you from having to click onto a folder to see its contents? E.g., in your example, as things currently are, clicking on `Project A` will show you the pinned dashboards at the top of the content area. But you're saying that maybe you're e.g. expanding a bunch of subfolders like `Sub-project A` and you don't want to have to navigate to that folder in order to see its pinned dashboards. Correct?

**Comment by mkopinsky** [comment]
Yes, exactly.

What's missing from my mockup is chevrons (idk if those made it into your final design) - I'm imagining that as I expand Project A, I immediately see (without navigating to a separate full page) that there are two subfolders and two pinned dashboards.

**Pull request merged**
